index.js

```diff
@@ -3,89 +3,115 @@
+    return awsPath;
+}
 
-        return nodefn.call(s3.putObject.bind(s3), {
-            ACL: 'public-read',
-            Bucket: options.bucket,
-            Key: targetFilename,
-            Body: buffer,
-            ContentType: image.type,
-            CacheControl: 'max-age=' + (30 * 24 * 60 * 60) // 30 days
-        });
-    })
-    .then(function(result) {
-        self.logInfo('ghost-s3', 'Temp uploaded file path: ' + image.path);
-    })
-    .then(function() {
-        return when.resolve(awsPath + targetFilename);
-    })
-    .catch(function(err) {
-        // fs.unlink(image.path, function (err) {
-        //     if (err) throw err;
-        //     errors.logInfo('ghost-s3', 'Successfully deleted temp file uploaded');
-        // });
-        self.logError(err);
-        throw err;
-    });
+function logError(error) {
+    console.log('error in ghost-s3', error);
 };
 
-// middleware for serving the files
-S3Store.prototype.serve = function() {
-    // a no-op, these are absolute URLs
-    return function (req, res, next) {
-      next();
-    };
+function logInfo(info) {
+    console.log('info in ghost-s3', info);
 };
 
-S3Store.prototype.getTargetDir = function() {
-    var MONTHS = [
-        'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
-        'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'
-    ];
-    var now = new Date();
-    return now.getFullYear() + '/' + MONTHS[now.getMonth()] + '/';
+function getTargetDir() {
+    var now = moment();
+    return now.format('YYYY/MM/');
 };
 
-S3Store.prototype.getTargetName = function(image, targetDir) {
-    var ext = path.extname(image.name),
-        name = path.basename(image.name, ext).replace(/\W/g, '_');
+function getTargetName(image, targetDir) {
+    var ext = path.extname(image.name);
+    var name = path.basename(image.name, ext).replace(/\W/g, '_');
 
     return targetDir + name + '-' + Date.now() + ext;
 };
 
-S3Store.prototype.logError = function(error) {
-    console.log('error in ghost-s3', error);
+function validOptions(opts) {
+    return (opts.accessKeyId && \
+        opts.secretAccessKey && \
+        opts.bucket && \
+        opts.region);
+}
+
+S3Store.prototype.save = function(image) {
+    if (!validOptions(options)) {
+      return Bluebird.reject('ghost-s3 is not configured');
+    }
+
+    var targetDir = getTargetDir();
+    var targetFilename = getTargetName(image, targetDir);
+
+    var s3 = new AWS.S3({
+        accessKeyId: options.accessKeyId,
+        secretAccessKey: options.secretAccessKey,
+        bucket: options.bucket,
+        region: options.region
+    });
+
+    return readFileAsync(image.path)
+        .then(function(buffer) {
+            var params = {
+                ACL: 'public-read',
+                Bucket: options.bucket,
+                Key: targetFilename,
+                Body: buffer,
+                ContentType: image.type,
+                CacheControl: 'max-age=' + (1000 * 365 * 24 * 60 * 60) // 365 days
```

Comment:
It would be nice to allow the user to configure this somehow, maybe also through the configs